Merge the intrinsic and user tests for `select_unpredictable`

tgross35 opened the pull request:
[1] mentions that having a single test with `-Zmerge-functions=disabled` is preferable to having two separate tests.  Apply that to the new `select_unpredictable` test here.

[1]: https://github.com/rust-lang/rust/pull/133964#issuecomment-2569693325

<!-- homu-ignore:start -->
r? @the8472
<!-- homu-ignore:end -->

tgross35 committed:
Merge the intrinsic and user tests for `select_unpredictable`

[1] mentions that having a single test with `-Zmerge-functions=disabled`
is preferable to having two separate tests.  Apply that to the new
`select_unpredicatble` test here.

[1]: https://github.com/rust-lang/rust/pull/133964#issuecomment-2569693325

the8472 commented on tests/codegen/intrinsics/select_unpredictable.rs:
```diff
@@ -33,3 +36,32 @@ pub fn test_zst(p: bool, a: (), b: ()) -> () {
+
+#[no_mangle]
+pub fn test_zst2(p: bool, a: (), b: ()) -> () {
+    // CHECK-LABEL: define{{.*}} @test_zst2
+    p.select_unpredictable(a, b)
```
this doesn't seem to test anything, just that the function exists, not what it does.

tgross35 replied:
I assume this was just intended to verify that no `select` is emitted, so I added a commit to assert the function body is empty.

the8472 reviewed (approved):
The change itself is fine, but one of the existing tests looks strange

tgross35 committed:
Expand the `select_unpredictable` test for ZSTs

For ZSTs there is no selection that needs to take place, so assert that
no `select` statement is emitted.

the8472 commented:
@bors r+

bors commented:
:pushpin: Commit 74d2d4bfa475a131df0c9ee0754b022a458cd152 has been approved by `the8472`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=the8472 74d2d4bfa475a131df0c9ee0754b022a458cd152 -->
<!-- homu: {"type":"Approved","sha":"74d2d4bfa475a131df0c9ee0754b022a458cd152","approver":"the8472","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 74d2d4bfa475a131df0c9ee0754b022a458cd152 with merge b3d730f172c580438f2e832e818a6a5179de4b45...
<!-- homu: {"type":"BuildStarted","head_sha":"74d2d4bfa475a131df0c9ee0754b022a458cd152","merge_sha":"b3d730f172c580438f2e832e818a6a5179de4b45"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/12627009342/job/35181052428)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/12627009342/job/35181052428","builder_name":"checks-actions"} -->

rust-log-analyzer commented:
A job failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/12627009342/job/35181004558) [(plain)](https://github.com/rust-lang-ci/rust/commit/b3d730f172c580438f2e832e818a6a5179de4b45/checks/35181004558/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
 62  694M   62  432M    0     0  77.2M      0  0:00:08  0:00:05  0:00:03 77.2M
 76  694M   76  530M    0     0  80.3M      0  0:00:08  0:00:06  0:00:02 86.8M
 93  694M   93  651M    0     0  85.7M      0  0:00:08  0:00:07  0:00:01 91.6M
100  694M  100  694M    0     0  88.1M      0  0:00:07  0:00:07 --:--:-- 97.8M
clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/mlir-reduce: Lzma library error: Corrupted input data
tar: Error exit delayed from previous errors.
##[error]Process completed with exit code 1.
[command]/usr/local/bin/git version
git version 2.47.1
Copying '/Users/runner/.gitconfig' to '/Users/runner/work/_temp/472d8406-5ebb-4a39-8b67-a0c22dd395a0/.gitconfig'
Temporarily overriding HOME='/Users/runner/work/_temp/472d8406-5ebb-4a39-8b67-a0c22dd395a0' before making global git config changes
```

</details>

tgross35 commented:
```text
Attempting with retry: curl -f https://ci-mirrors.rust-lang.org/rustc/clang%2Bllvm-15.0.7-x86_64-apple-darwin21.0.tar.xz -o clang+llvm-15.0.7-x86_64-apple-darwin21.0.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
...
100  694M  100  694M    0     0  88.1M      0  0:00:07  0:00:07 --:--:-- 97.8M
clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/mlir-reduce: Lzma library error: Corrupted input data
```

@bors retry

bors commented:
:hourglass: Testing commit 74d2d4bfa475a131df0c9ee0754b022a458cd152 with merge 243d2ca4db6f96d2d18aaf3a2381251d38eb6b0b...
<!-- homu: {"type":"BuildStarted","head_sha":"74d2d4bfa475a131df0c9ee0754b022a458cd152","merge_sha":"243d2ca4db6f96d2d18aaf3a2381251d38eb6b0b"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/12631179690/job/35199469664)
Approved by: the8472
Pushing 243d2ca4db6f96d2d18aaf3a2381251d38eb6b0b to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"the8472","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/12631179690/job/35199469664"},"merge_sha":"243d2ca4db6f96d2d18aaf3a2381251d38eb6b0b"} -->

rust-timer commented:
Finished benchmarking commit (243d2ca4db6f96d2d18aaf3a2381251d38eb6b0b): [comparison URL](https://perf.rust-lang.org/compare.html?start=13738b0abeeb6b7b8b52cc16aa8ece267b58f017&end=243d2ca4db6f96d2d18aaf3a2381251d38eb6b0b&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=13738b0abeeb6b7b8b52cc16aa8ece267b58f017&end=243d2ca4db6f96d2d18aaf3a2381251d38eb6b0b&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=13738b0abeeb6b7b8b52cc16aa8ece267b58f017&end=243d2ca4db6f96d2d18aaf3a2381251d38eb6b0b&stat=max-rss)
<details>
<summary>Results (primary -0.9%, secondary -2.2%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | 3.6%  | [3.6%, 3.6%]   | 1     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -1.9% | [-2.6%, -0.6%] | 5     |
| Improvements ✅ <br /> (secondary) | -2.2% | [-2.2%, -2.2%] | 1     |
| All ❌✅ (primary)                 | -0.9% | [-2.6%, 3.6%]  | 6     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=13738b0abeeb6b7b8b52cc16aa8ece267b58f017&end=243d2ca4db6f96d2d18aaf3a2381251d38eb6b0b&stat=cycles:u)
<details>
<summary>Results (primary -4.1%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -4.1% | [-5.3%, -3.5%] | 6     |
| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
| All ❌✅ (primary)                 | -4.1% | [-5.3%, -3.5%] | 6     |
</details>

### [Binary size](https://perf.rust-lang.org/compare.html?start=13738b0abeeb6b7b8b52cc16aa8ece267b58f017&end=243d2ca4db6f96d2d18aaf3a2381251d38eb6b0b&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 764.087s -> 763.649s (-0.06%)
**Artifact size**: 325.66 MiB -> 325.68 MiB (0.00%)
